src/App.tsx

```diff
@@ -1,93 +1,113 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useCallback, useMemo } from 'react';
+import { useLocation } from 'react-router-dom';
+import { useLocalStorage } from './hooks/useLocalStorage';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { Status } from './types/Status';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useLocalStorage('todos', []);
+
+  const location = useLocation();
+
+  const handleAddTodo = (title: string) => {
+    const newTodo = {
+      id: +new Date(),
+      title,
+      completed: false,
+    };
+
+    setTodos([...todos, newTodo]);
+  };
+
+  const handleToggleCompleted = (todoId: number) => {
+    const updateTodoList = todos.map((todo:Todo) => (
+      todo.id === todoId ? { ...todo, completed: !todo.completed } : todo
+    ));
+
+    setTodos(updateTodoList);
+  };
+
+  const handleToggleAll = () => {
+    const notCompletedAll = todos.every((todo:Todo) => todo.completed);
+    const changeAll = todos.map((todo:Todo) => (
+      { ...todo, completed: !notCompletedAll }));
+
+    setTodos(changeAll);
+  };
+
+  const handleRemoveTodo = useCallback((todoId: number) => {
+    const newsTodos = todos.filter((todo:Todo) => todo.id !== todoId);
+
+    setTodos(newsTodos);
+  }, [todos]);
+
+  const handleClearCompleted = () => {
+    const deleteAllCompleted = todos.filter((todo:Todo) => !todo.completed);
```

Comment:
I'd rename it to smth like `activeTodos`
`deleteAllCompleted` sounds like it's a function that deletes todo

Just FYI (task is approved)